### ISSUE TITLE
Fragment of interface generates intersection type

### ISSUE THREAD

**Issue by kgtkr** [dotansimha/graphql-code-generator#1533]
**Describe the bug**

**To Reproduce**

1. Use '...' as schema
```
interface Base{
  id:ID!
}

interface A implements Base{
  id:ID!
  x:Int!
}

interface B implements Base{
  id:ID!
  y:Int!
}
```
2. Use '...' as documents
```
fragment base on Base{
  ...a
  ...b
}

fragment a on A{
  id
  a
}

fragment a on B{
  id
  b
}
```
3. Run the following command: `graphql-codegen ...`

**Expected behavior**

```ts
type BaseFragment=AFramgent|BFragment;
```

Generated code

```ts
type BaseFragment=AFramgent&BFragment;
```

**Schema/Documents**
If possible and relevant, please provide your GraphQL schema and GraphQL documents.

**Environment:**

- OS:
- Codegen:
- Node:

**Additional context**

**Comment by kgtkr** [comment]
Relation(?) #695

**Comment by dotansimha** [comment]
@kgtkr Which version do you use?

**Comment by kgtkr** [comment]
1.0.2

**Comment by dotansimha** [comment]
@kgtkr thanks, fixing it in: https://github.com/dotansimha/graphql-code-generator/pull/1534

**Comment by dotansimha** [comment]
Fixed in 1.0.3. @kgtkr
